finagle-core/src/main/scala-2.13+/com/twitter/finagle/Dtab.scala

```diff
@@ -0,0 +1,33 @@
+package com.twitter.finagle
+
+import scala.collection.immutable.VectorBuilder
+import scala.collection.mutable.ReusableBuilder
+
+/**
+ * A Dtab--short for delegation table--comprises a sequence of
+ * delegation rules. Together, these describe how to bind a
+ * [[com.twitter.finagle.Path]] to a set of
+ * [[com.twitter.finagle.Addr]]. [[com.twitter.finagle.naming.DefaultInterpreter]]
+ * implements the default binding strategy.
+ *
+ * @see The [[https://twitter.github.io/finagle/guide/Names.html#interpreting-paths-with-delegation-tables user guide]]
+ *      for further details.
+ */
+case class Dtab(dentries0: IndexedSeq[Dentry]) extends DtabBase
+
+object Dtab extends DtabCompanionBase {
+  
+}
+
+final class DtabBuilder extends ReusableBuilder[Dentry, Dtab] {

```

Comment:
can we document the distinction between the 2.12 stuff vs the 2.13 stuff?  would we be able to just use this if we didn't need to build for 2.11, for example?

Reply:
The 2.12 version has `CanBuildFrom`, and 2.13 doesn't. Without it you will never be able to map a `Dtab` with a `Dentry => Int` and get back a `BitSet`.

It's a bit of a niche concern I suppose. You could consider breaking downstream and just have a Dentry => Dentry overload of map, and if 2.13 was all we ever knew and this was greenfields, it would probably be a good idea.  Whether it's enough of a justification to break downstream if they are doing something I consider moderately silly is really not up to me.

Reply:
right.  I'm wondering if we can add a comment to this file that explains why we have to make this change?  it sounds like behavior is changing incompatibly when we switch to 2.13.x anyway, so maybe the TODO is to make the breaking change in <= 2.12.x to unify the builds and prepare users for 2.13.x?  we could add that as a comment too.

Reply:
I'll try to make this a bit neater, and make some compilation and run tests to demonstrate what behaviour is unified (and document what is and isn't). For an integrated collection in 2.13 the companion should extend `Factory` (among other reasons?) so that `Iterable#to(Dtab)` works, and that should also work on 2.12 with the scala.collection.compat library.